src/main/kotlin/com/atlassian/performance/tools/aws/api/Investment.kt

```diff
@@ -18,12 +18,14 @@ import com.amazonaws.services.ec2.model.Tag as Ec2Tag
  * @param disposable if true, then the investment can be cancelled even before [lifespan] and get its cost reduced
  * @param reuseKey if a new investment would have the same reuse key as an existing investment, the old one can be
  *                 reused, reducing the cost
+ * @param resourceOwner AWS resource owner, defaults to current Atlassian JPT maintainer
  */
 data class Investment(
     private val useCase: String,
     val lifespan: Duration,
     private val disposable: Boolean = true,
-    val reuseKey: () -> String = { "jpt-${UUID.randomUUID()}" }
+    val reuseKey: () -> String = { "jpt-${UUID.randomUUID()}" },
+    private val resourceOwner: String = "jforemski"
```

Comment:
This class needs builder. I believe the change doubles the constructors count from 4 to 8 and all of them are part of lib's API.